test/e2e/ocpbugs48050_test.go

```diff
@@ -0,0 +1,611 @@
+}
+
+// TestOCPBUGS48050 validates the new metric [1],
+// duplicate_te_header_total, which detects duplicate
+// Transfer-Encoding headers per backend response. This test addresses
+// issues encountered when upgrading from HAProxy 2.2 in OpenShift
+// 4.13 to HAProxy 2.6 in OpenShift 4.14, where responses are rejected
+// due to RFC 7230 compliance.
+//
+// [1] https://github.com/openshift/router/pull/626.
+func TestOCPBUGS48050(t *testing.T) {
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ns, err := setupOCPBUGS48050(t, 10)
+	if err != nil {
+		t.Fatalf("failed to setup test resources: %v", err)
+	}
+
+	routeList, err := waitForAllRoutesAdmitted(ns.Name, 2*time.Minute, func(admittedRoutes, totalRoutes int, pendingRoutes []string) {
+		if len(pendingRoutes) > 0 {
+			t.Logf("%d/%d routes admitted. Waiting for: %s", admittedRoutes, totalRoutes, strings.Join(pendingRoutes, ", "))
+		} else {
+			t.Logf("All %d routes in namespace %s have been admitted", totalRoutes, ns.Name)
+		}
+	})
+	if err != nil {
+		t.Fatalf("Error waiting for routes to be admitted: %v", err)
+	}
+
+	httpClient := &http.Client{
+		Timeout: time.Minute,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	httpGet := func(url string, check func(*http.Response, error) error) error {
+		t.Logf("GET %s", url)
+		resp, err := httpClient.Get(url)
+		if resp != nil {
+			defer func(Body io.ReadCloser) {
+				_ = Body.Close()
+			}(resp.Body)
+		}
+		return check(resp, err)
+	}
+
+	routeIndex := func(name string) int {
+		parts := strings.Split(name, "-")
+		if len(parts) < 2 {
+			t.Fatalf("Unexpected route name format: %q. Expected at least two parts separated by '-'", name)
+		}
+
+		index, err := strconv.Atoi(parts[len(parts)-1])
+		if err != nil {
+			t.Fatalf("Failed to convert route index to integer for route %q: %v", name, err)
+		}
+
+		return index
+	}
+
+	for _, route := range routeList.Items {
+		host := route.Spec.Host
+
+		// Retry only the /healthz endpoint because sometimes
+		// we get a 503 status even though the routes have
+		// been admitted.
+		if err := wait.Poll(5*time.Second, 1*time.Minute, func() (bool, error) {
+			healthzURL := fmt.Sprintf("http://%s/healthz", host)
+			if err := httpGet(healthzURL, func(resp *http.Response, err error) error {
+				if err != nil {
+					t.Logf("Health check error for route %s: %v. Retrying...", host, err)
+					return err
+				}
+				if resp.StatusCode != http.StatusOK {
+					t.Logf("Unexpected status code for %s: %d. Retrying...", healthzURL, resp.StatusCode)
+					return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+				}
+				return nil
+			}); err != nil {
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			t.Fatalf("/healthz checks failed for route %s: %v", host, err)
```

Comment:
No, we only call /single-te and /duplicate-te an upfront known quantity of times - that's how we can assert the number of hits to the metric.

Reply:
The purpose of the /healthz check is to address the issue where, even after the route is admitted, you might still encounter a 503 status when hitting the /single-te or /duplicate-te endpoints. I experienced this problem yesterday. By successfully hitting the /healthz endpoint for each route first, we ensure a more reliable state before running the actual tests.

Reply:
> The purpose of the /healthz check is to address the issue where, even after the route is admitted, you might still encounter a 503 status when hitting the /single-te or /duplicate-te endpoints. I experienced this problem yesterday. By successfully hitting the /healthz endpoint for each route first, we ensure a more reliable state before running the actual tests.

I agree with the approach of hitting `/healthz` before `/single-te` and `/duplicate-te`. What I meant is if we set `t.Errorf()` for `/healthz` check we can still let the other routes to try `/healthz`-`/single-te`-`/duplcate-te` sequence instead of stopping the test on the first route to fail `/healthz` probe.

Reply:
What’s the benefit of allowing the test to continue once it has already failed? We’ve pinged the endpoint for 1 minute (@5s intervals) without success, so it makes sense to exit early since the test is going to fail regardless.

Reply:
> What’s the benefit of allowing the test to continue once it has already failed?

The same benefit as [you mentioned here](https://github.com/openshift/cluster-ingress-operator/pull/1150#discussion_r1782937175).

Reply:
I want to fail early on /healthz. I consider this part of the setup, similar to the route, service, deployment, or namespace not being created. If the test setup cannot complete successfully, we should bail as soon as possible.

Reply:
> > What’s the benefit of allowing the test to continue once it has already failed?
> 
> The same benefit as [you mentioned here](https://github.com/openshift/cluster-ingress-operator/pull/1150#discussion_r1782937175).

I do what I'm saying is a contradiction. But I'd argue once we’ve reached the end of the test, there are no other loops, timeouts, or retries. We’re simply comparing expected versus actual results, and the time to make that comparison is negligible compared to retrying operations that typically take many minutes. The test is effectively complete, we're just reporting final status.

Reply:
Look at it another way: if every health check fails, we’d consume 30 minutes in total (3 termination types × 10 routes × 1 minute per route). This is why I’m considering hitting the /healthz setup code—if there’s no service, we cannot continue. I’m arguing for the same behaviour with /healthz.